llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -3321,6 +3325,29 @@ bool AMDGPUInstructionSelector::selectBufferLoadLds(MachineInstr &MI) const {
   return constrainSelectedInstRegOperands(*MIB, TII, TRI, RBI);
 }
 
+bool AMDGPUInstructionSelector::selectBufferLoadTfe(MachineInstr &MI) const {
+  // Get an iterator for the current MI
+  MachineBasicBlock *MBB = MI.getParent();
+  auto MII = MachineBasicBlock::iterator(MI);
+  MII = std::next(MII);
+
+  // Standard selection works fine, use that to generate the instruction
+  if (!selectImpl(MI, *CoverageInfo))
+    return false;
```

Comment:
This is pretty ugly and you're making assumptions about what the generated selector does. You don't have the equivalent handling in the DAG path? I assume this is done in the post-isel hook. FinalizeISel still runs the same EmitInstrWithCustomInserter, so the same code should work for both paths without this extra post-processing?

Reply:
Good idea - I've re-factored to use EmitInstrWithCustomInserter.